### PR TITLE
Detect cpu stuck bsc#1052258 and collect logs

### PR DESCRIPTION
**Pull request by rwx788**
As result of investigation of
[poo#30085](https://progress.opensuse.org/issues/30085) we have
identified that problem is not directly with the test, but that's not
obvious from how it looks. That's the best thing I believe we can do to
point to the bug so it's visible. Seems that it's already fixed in newer
kernel versions, hence not much sense in investigation.

[Verification run](http://g226.suse.de/tests/1228).

### CODE REVIEW

**Comment by foursixnine** [comment]
@rwx788 Since we're waiting for #4672 that should be ready for merging next week... and will include [poo#28027](https://progress.opensuse.org/issues/28027). You can get a head start and work to use that method instead that would help a bit in this scenarios

**Comment by rwx788** [comment]
@foursixnine oh, that looks great. Thanks for the info, then I can try it out!

File: tests/update/updates_packagekit_gpk.pm [hunk 1]
```diff
@@ -14,13 +14,20 @@ use base "x11test";
...
-sub turn_off_screensaver {
+sub turn_off_screensaver
+ {
+    my ($self) = @_;
     # Turn off screensaver
     x11_start_program('xterm');
     become_root;
     ensure_serialdev_permissions;
+    # Trying to detect bsc#1052258 which leads to the test failure occasionally
+    if(is_leap '42.3' && !script_run "dmesg | grep -P soft lockup - CPU#\\d+ stuck") {
```
Comment (kravciak) [inline tests/update/updates_packagekit_gpk.pm]: You need parentheses here, your conditoin is equal to:
```perl
if(is_leap ('42.3' && !script_run "dmesg....")) {
```
Which bubbles into `is_leap(0)` or `is_leap(1)`

Comment (okurz) [reply tests/update/updates_packagekit_gpk.pm]: If a CPU is stuck I doubt you would be able to call the command but we might be lucky :)

**Comment by rwx788** [comment]
So it didn't work well with new api call, I was not able to make it work. Consequently, this improvement is scheduled for later, but I would propose initial solution.

File: tests/update/updates_packagekit_gpk.pm [hunk 2]
```diff
@@ -14,13 +14,23 @@ use base "x11test";
...
     ensure_serialdev_permissions;
+
+    # Trying to detect bsc#1052258 which leads to the test failure occasionally on Leap 42.3
+    if (is_leap('=42.3') && script_run("dmesg | grep -P 'soft lockup - CPU#\\d+ stuck'")) {
+        # Fail on cpu stuck, see bsc#1052258 and collect logs if machine still responds
+        record_soft_failure 'cpu soft lockup detected, see bsc#1052258';
+        $self->save_upload_y2logs;
+        select_console 'x11';    # Trying to continue with the test
+    }
+
```
Comment (okurz) [inline tests/update/updates_packagekit_gpk.pm]: can you comment why this code should be here where it is kinda not well fitting? I mean the method is called `turn_off_screensaver` so there should be at least a good name explaining why we have this here.

File: tests/update/updates_packagekit_gpk.pm [hunk 3]
```diff
@@ -14,13 +14,23 @@ use base "x11test";
...
+    my ($self) = @_;
     # Turn off screensaver
     x11_start_program('xterm');
     become_root;
     ensure_serialdev_permissions;
+
+    # Trying to detect bsc#1052258 which leads to the test failure occasionally on Leap 42.3
+    if (is_leap('=42.3') && script_run("dmesg | grep -P 'soft lockup - CPU#\\d+ stuck'")) {
+        # Fail on cpu stuck, see bsc#1052258 and collect logs if machine still responds
+        record_soft_failure 'cpu soft lockup detected, see bsc#1052258';
```
Comment (okurz) [inline tests/update/updates_packagekit_gpk.pm]: I assume you have already seen this problem in recent tests. But the last comment in the bug was from 2017-10-29 and stating that the problem is not there anymore so please comment on the bug or explain what this PR has to do with the bug then

Comment (asmorodskyi) [reply tests/update/updates_packagekit_gpk.pm]: but the bug is in state NEW , probably developer not sure enough to close it ?
I think it is a good catcher for this bug.
@rwx788 +1

File: tests/update/updates_packagekit_gpk.pm (hunk 2 above)
Comment (asmorodskyi) [reply tests/update/updates_packagekit_gpk.pm]: and here I agree with you , probably we need to put it into some more reasonable place

**Comment by asmorodskyi** [comment]
@rwx788  I would vote for separate module , which we can insert at the end of some long tests .
This module will grep dmesg for errors . and will hold list of known errors and one which we constatly looking for . Not just this bug I will come to you to discuss this idea

**Comment by kravciak** [comment]
@asmorodskyi - your idea already exists ;) https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/caasp/journal_check.pm

I guess this is PR is solution for more specific problem.

File: tests/update/updates_packagekit_gpk.pm (hunk 3 above)
Comment (rwx788) [reply tests/update/updates_packagekit_gpk.pm]: Bug is in the version kernel which we have in leap 42.3, which we use to verify upgrade scenarios. With other kernel versions it's not there. I'll put this in the code comment too.

**Comment by rwx788** [comment]
@asmorodskyi there already couple of features, including one which @foursixnine has suggested. I've spent already more than required on this thing which is optional in the end, so I guess in case it's not good enough, I'll close it and create separate ticket. In the end it's just a hint, so we don't get ticket about broken test, when it's actually product bug...

**Comment by kravciak** [comment]
PR looked good from my POV - in case you change your mind ;)

**Comment by rwx788** [comment]
@kravciak thanks, but me myself would make it work with new api call, @dasantiago kindly contacted me regarding it, but we could not figure out why it didn't work, so we'll continue in [poo#34006](https://progress.opensuse.org/issues/34006) if it will be considered as valuable enough.

**Pull request closed without merging**